slack-lint/src/main/java/slack/lint/resources/ImportAliasesLoader.kt

```diff
@@ -0,0 +1,82 @@
+package slack.lint.resources
+
+import com.android.tools.lint.detector.api.Context
+import java.io.File
+import java.io.StringReader
+import java.util.Properties
+
+object ImportAliasesLoader {
+    const val PROPERTY_FILE = "gradle.properties"
+    const val IMPORT_ALIASES_PROPERTY = "nonTransitiveRClass.importAliases"
+
+    private const val TOP_LEVEL_FILENAME = "settings.gradle"
+
+    // Caches the import aliases to improve performance.
+    var importAliases = emptyMap<String, String>()
+        private set
+
+    /**
+     * Loads the import aliases from the root [PROPERTY_FILE].
+     * If it can't find it, it defaults to the local [PROPERTY_FILE].
+     */
+    fun loadImportAliases(context: Context) {
+        // Project#getPropertyFiles returns the current module's PROPERTY_FILE file.
+        // To avoid having to define IMPORT_ALIASES_PROPERTY in every module's PROPERTY_FILE,
+        // this loads it from the root folder's PROPERTY_FILE.
+
+        if (importAliases.isEmpty()) {
+            val root: File? = findRoot(context)
+
+            File(root, PROPERTY_FILE).run {
+                if (this.exists()) {
+                    val aliasesProperty = getProperty(context, IMPORT_ALIASES_PROPERTY)
+                    if (aliasesProperty != null)
+                        loadFromProperty(aliasesProperty)
```

Comment:
We shouldn't be grepping from files other than those offered by `context.project.propertyFiles` because it would interfere with gradle caching, please just use the `context.project.propertyFiles` API.

Reply:
Removing this would force us to [define nonTransitiveRClass.importAliases in every module's gradle.properties](451f2d67c132a2e70dbfb8ee80b527b2724bc3a5) file and then figure out a way to sync those whenever one is updated so that the list isn't empty when run from the IDE or `module:lintAnalyzeRelease` though. 
Is there any other way, that you know of, to get the root or app `gradle.properties` without breaking configuration to avoid ☝️ ?

Reply:
that link doesn't work fyi. Does it not give you the root gradle.properties file in that API though?

Reply:
hmm, I see that it's not. Urg.

Let me try something and report back

Reply:
For posterity - there's an API in lint for this now we can use, example in https://github.com/slackhq/slack-lints/pull/28

Reply:
Switched to using `StringOption` in abf3d4c56ce5deb731d85499d3a300befc515ca5.